[AWS] Set everything as internal in AWS

surminus opened the pull request:
This change comes off the back of a discussion that was had in a [PR that I opened for Plek][1]. It looks to solve an issue where we initially tried to override specific cases of Plek lookups, but kept running into small edge cases.

Set "GOVUK_APP_DOMAIN" as internal by default. Ensure that each application sets a Plek override for itself so that if it ever references itself, it redirects the user to the correct location.

This hopefully solves a bunch of problems we have come across when trying to correctly route traffic internally. We had an issue where applications would misuse Plek to find out what their URL is to redirect users.

[1]: https://github.com/alphagov/plek/pull/54#issuecomment-353390978

boffbowsh commented on modules/govuk/manifests/deploy/config.pp:
```diff
@@ -92,31 +92,27 @@
+      # We do not host Licensify in AWS, so need to redirect to
+      # it's external domain
+      'PLEK_SERVICE_LICENSIFY_URI': value => "https://licensify.${licensify_app_domain}";
+      # Apps that auth with Signon should do so across the publishing domain
```
Would be good to have the reason documented here. I can't think of it off the top of my head

boffbowsh commented on modules/govuk/spec/defines/govuk__app__config_spec.rb:
```diff
@@ -92,6 +92,45 @@
+    end
+
+    context 'in AWS with hyphenated app name' do
+      let(:title) { 'bella-the-cat' }
```
😸

surminus commented:
I updated the comment, but I'm still unsure on the specific reason it happened. 

The behaviour was that you would login to Signon (after updating your `/etc/hosts` so that `integration.publishing.service.gov.uk` pointed to the AWS environment), try to click through to an application and be presented with a failed DNS lookup for `integration.govuk-internal.digital`. This change resolved the issue, but I am still unsure exactly where or how the applications or Signon were using Plek to set this URL.

boffbowsh commented:
Is that not fixed by Signon having an override set for itself? I *think* that's to do with signon having to set a callback URL for itself. Or was it e.g. `publisher.integration.govuk-internal.digital` that it was trying to send you to?

surminus commented:
That's correct. I will test without this change and see if it still persists.

surminus commented:
@boffbowsh I've added the commit back in. After testing by browsing to an app, I was redirected to Signon using the incorrect URL which resulted in the following error:

```
This site can’t be reached

signon.integration.govuk-internal.digital’s server DNS address could not be found.
Try running Network Diagnostics.
DNS_PROBE_FINISHED_NXDOMAIN
```

I believe this is because the applications (or gds-sso) use Plek to find the Signon URL to redirect users to, so we need to ensure it is set to the Publishing domain.

I've added it as a separate commit with more detail, and also set comments so they are clearer.
